### PR TITLE
Clean up the filesystem stuff and add support for HF ckpt downloading

### PR DESCRIPTION
**Pull request by gpetters-amd**
This works fine for SD2.1 but I've noticed an issue downloading the safetensors file for clip for SD1.5. There's still some more cleanup necessary here, so I'd appreciate a second pair of eyes on it. I'm also not sure if the way I'm centralizing all the filesystem logic (just using the args and calling `os.path.join` at each usage) is the right thing to do or if we want to have everything go through, say, functions in `file_utils.py`.

### CODE REVIEW

File: apps/shark_studio/web/ui/sd.py [hunk 1]
```diff
@@ -392,6 +397,9 @@ def base_model_changed(base_model_id):
                         elem_id="negative_prompt_box",
                         show_copy_button=True,
                     )
+                    with gr.Row():
```
Comment (monorimet) [inline apps/shark_studio/web/ui/sd.py:400]: Why do we add an explicit option to save the prompt? It should write prompts to the json config, so this seems a bit redundant to me.

File: apps/shark_studio/web/ui/sd.py [hunk 2]
```diff
@@ -631,19 +624,18 @@ def base_model_changed(base_model_id):
...
+                            default_config_file = os.path.join(
+                                get_configs_path(),
+                                "default_sd_config.json",
+                            )
+                            write_default_sd_config(default_config_file)
+                            sd_json = gr.JSON(
+                                label="SD Config",
+                                elem_classes=["fill"],
+                                value=view_json_file(default_config_file),
+                                render=False,
```
Comment (monorimet) [inline apps/shark_studio/web/ui/sd.py:637]: Why turn off rendering this?

**Comment by monorimet** [comment]
Thanks @gpetters-amd. I left a few comments, but looks better for the most part.

File: apps/shark_studio/web/ui/sd.py (hunk 2 above)
Comment (gpetters-amd) [reply apps/shark_studio/web/ui/sd.py:637]: It's the swap the order - I moved the render call to line 702 below.

File: apps/shark_studio/web/ui/sd.py (hunk 1 above)
Comment (gpetters-amd) [reply apps/shark_studio/web/ui/sd.py]: Yeah, the new config stuff is half-baked. Took it out, we can discuss how to handle it after we've got stable out.

**Review by monorimet — APPROVED** [review]
Looks better. Thank you.

**Pull request merged**